samples/drivers/led_lp50xx/README.rst

```diff
@@ -1,12 +1,12 @@
-.. _lp503x:
+.. _lp50xx:
 
 LP5030/6: 10 or 12 RGB channels
```

Comment:
The title above and the text in the `Building and Running` section must be updated too.

In addition the string lp503x can still be found a lot. Please fix that.

```
$ grep -ric lp503x samples/drivers/led_lp50xx
samples/drivers/led_lp50xx/src/main.c:32
samples/drivers/led_lp50xx/CMakeLists.txt:1
samples/drivers/led_lp50xx/README.rst:2
samples/drivers/led_lp50xx/boards/lpcxpresso11u68.overlay:0
samples/drivers/led_lp50xx/sample.yaml:2
samples/drivers/led_lp50xx/prj.conf:0
```

Reply:
Hi @simonguinot , thank you for your reviews and also for your patience. Should I change also name and defines in lp503x.h file (include/zephyr/drivers/led/lp503x.h)? It's used only in the sample.

Reply:
Well I forgot about this header and it is annoying...

The `include/zephyr/drivers/led/lp503x.h` file is important. It provides the channel mapping to user application. And in turn the lp50xx sample provides an usage example. Since the mapping is different between the LP50{09,12` and LP50{30,36} devices, then this file must be updated to support both. For example, I can see that `LP503X_NUM_CHANNELS` and `LP503X_LED_COL_CHAN_BASE` should have a different value depending on the device type.

I think you have two options:

1. Make all the channel definitions and macros in `include/zephyr/drivers/led/lp503x.h` (to be renamed into `lp50xx.h`) conditional to the chip type (like you did in the driver itself).
2. Or add some static channel definitions and macros for the LP50{09,12} devices.

But in any case, the led_lp50xx sample must be compatible with all devices types (LP50{09,12} and LP50{30,36}). So in the code, you have to match the DT compatible to identify the device and then to use the proper channel mapping.

In addition, please test the sample with your device. Since `LP503X_LED_COL_CHAN_BASE` is wrong for the LP50{09,12}, then you should have noticed earlier that the sample is broken...

Reply:
I wasn't aware that we want this sample to work also with LP50{09,12}, as you can see in lpcxpresso11u68.overlay file I changed compatible to `compatible = "ti,lp5030", "ti,lp50xx";`, so it should work as before. It's also noticed in README.rst file.
I'm not sure, but doesn't option no. 2 make more sense? Do you assume creating similar file for LP50{09.12} chips, with proper name and macros? In this case probably sample code may look a little bit worse, but it seems like more convenient for future use for me.
Thank you for your help, time and answer 👍

Reply:
Okay, I'll try to do it better :)
I'm using custom board with LP5009 soldered, so it's not possible this way.